src/mkindex_algo.hpp

```diff
@@ -636,28 +642,15 @@ auto generateIndex(TStringSet                       & seqs,
                    LambdaIndexerOptions       const & options)
 {
     using TRedAlph       = seqan3::range_innermost_value_t<TStringSet>;
-
-    constexpr auto is_collection = seqan3::text_layout::collection;
-    using TSpec = IndexSpec<seqan3::alphabet_size<TRedAlph>>;
-    using TIndex = std::conditional_t<is_bi,
-                                      seqan3::bi_fm_index<TRedAlph, is_collection, TSpec>,
-                                      seqan3::fm_index<TRedAlph, is_collection, TSpec>>;
+    using TIndexSpec     = IndexSpec<seqan3::alphabet_size<TRedAlph>>;
+    using TIndex         = std::conditional_t<is_bi,
+                                             fmindex_collection::BiFMIndex<TIndexSpec>,
+                                             fmindex_collection::ReverseFMIndex<TIndexSpec>>;
```

Comment:
So `FMIndex` is the normal FMIndex which allows backward searches. A cursor always points beginning of a search query.

`ReverseFMIndex` reverses the text before building the index. This means it will allow only forward searches, with the downside that internally cursors now point to the end of the pattern. (This is compensated by carrying an extra cursor depth/query length variable to correct for this).